docroot/modules/custom/yukon_migrate/yukon_migrate.module

```diff
@@ -15,92 +15,63 @@ use Drupal\migrate\Plugin\MigrationInterface;
-  $migration_nids = [
-    'yukon_migrate_engagement',
-    'yukon_migrate_in_page_alert',
-    'yukon_migrate_basic_page',
-    'yukon_migrate_basic_page_translations',
-  ];
-  $migration_tids = [
-    'yukon_migrate_department',
-    'yukon_migrate_engagement_categories',
-    'yukon_migrate_media_folders',
-    'yukon_migrate_teams',
-  ];
-  $node_migration = in_array($migration->id(), $migration_nids);
-  $taxonomy_migration = in_array($migration->id(), $migration_tids);
+  $migration_type = strpos($source->getPluginId(), 'node') ? 'node' : NULL;
+  if (empty($migration_type)) {
+    $migration_type = strpos($source->getPluginId(), 'term') ? 'taxonomy_term' : NULL;
+  }
 
   /** @var \Drupal\Core\Database\Connection $database */
   $database = $source->getDatabase();
 
   // Get and set into source property Pathauto and Moderation State.
-  if ($node_migration) {
-    $nid = $row->getSourceProperty('nid');
-    // Revision ID.
-    $vid = $row->getSourceProperty('vid');
-
-    $query = $database->select('node', 'n')
-      ->fields('n', ['nid', 'vid'])
-      ->fields('ps', ['pathauto'])
-      ->fields('w', [
-        'from_state',
-        'state',
-      ]);
-    $query->condition('ps.entity_type', 'node');
-    $query->condition('n.nid', $nid);
-    $query->leftJoin('pathauto_state', 'ps', '[n].[nid] = [ps].[entity_id]');
-    $query->leftJoin('workbench_moderation_node_history', 'w', '[n].[nid] = [w].[nid]');
-    $result = $query->execute()->fetchAssoc();
-
-    if (!empty($result)) {
-      yukon_migrate_set_pathauto_value($result, $row);
+  if (!empty($migration_type)) {
+    $id = $row->getSourceProperty('tid');
+    if (empty($id)) {
+      $id = $row->getSourceProperty('nid');
+      $vid = $row->getSourceProperty('vid');
     }
 
-    $workbench_moderation = $query->condition('n.vid', $vid)
-      ->orderBy('stamp', 'DESC')
-      ->execute()
-      ->fetchAssoc();
+    if (!empty($id)) {
+      $query = $database->select('pathauto_state', 'ps')
+        ->fields('ps', ['pathauto', 'entity_id'])
+        ->fields('w', [
+          'from_state',
+          'state',
+        ]);
+      $query->condition('ps.entity_type', $migration_type);
+      $query->condition('ps.entity_id', $id);
+      $query->leftJoin('workbench_moderation_node_history', 'w', '[ps].[entity_id] = [w].[nid]');
+      $result = $query->execute()->fetchAssoc();
 
-    if (!empty($workbench_moderation)) {
-      // Get and set workbench moderation status.
-      $row->setSourceProperty('moderation_state', [
-        'current' => $workbench_moderation['state'],
-        'state' => $workbench_moderation['from_state'],
-      ]);
-    }
-  }
+      if (!empty($result)) {
+        yukon_migrate_set_pathauto_value($result, $row);
+      }
 
-  // Get and set into source property Pathauto.
-  if ($taxonomy_migration) {
-    $tid = $row->getSourceProperty('tid');
+      // Get and set workbench moderation status.
+      if (!empty($vid)) {
```

Comment:
It's complicated to understand what is $vid in this particular case. Can we make this part more clear, for example, add a check for that entity type === 'node'?